scapy/contrib/rpl.py

```diff
@@ -0,0 +1,294 @@
+# This file is part of Scapy.
+# See http://www.secdev.org/projects/scapy for more information.
+#
+# Scapy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Scapy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Scapy.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Copyright (C) 2020 Rahul Jadhav <nyrahul@gmail.com>
+
+# RFC 6550
+# scapy.contrib.description = Routing Protocol for LLNs (RPL)
+# scapy.contrib.status = loads
+
+from scapy.packet import Packet
+from scapy.fields import ByteEnumField, ByteField, IP6Field, ShortField, \
+    XShortField, BitField, BitEnumField, FieldLenField, StrLenField, IntField
+from scapy.layers.inet6 import icmp6rplcodes, RPL, icmp6ndraprefs, \
+    _IP6PrefixField
+
+
+# https://www.iana.org/assignments/rpl/rpl.xhtml#mop
+rplmop = {0: "No Downward routes",
+          1: "Non-Storing",
+          2: "Storing with no multicast support",
+          3: "Storing with multicast support",
+          4: "P2P Route Discovery"}
+
+
+# https://www.iana.org/assignments/rpl/rpl.xhtml#control-message-options
+rploptsstr = {0: "Pad1",
+              1: "PadN",
+              2: "DAG Metric Container",
+              3: "Routing Information",
+              4: "DODAG Configuration",
+              5: "RPL Target",
+              6: "Transit Information",
+              7: "Solicited Information",
+              8: "Prefix Information Option",
+              9: "Target Descriptor",
+              10: "P2P Route Discovery"}
+
+
+rplopts = {
+}
+
+
+class RPLGuessOption:
+    name = "Dummy RPL Option class that implements guess_payload_class()"
+
+    def guess_payload_class(self, p):
```

Comment:
Instead of using `guess_payload_class`, we prefer to use multiple `bind_layers` calls.

You can keep the dummy and do all your `bind_layers` call on it.

Reply:
Fixed: `bind_layers` is actually better to use. More readable.